src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/converters/FlowRulesConverter.java

```diff
@@ -69,13 +71,41 @@ public InstallSpeakerCommandsRequest buildFlowInstallCommand(SwitchId switchId,
                 .build();
     }
 
+    /**
+     * Build a list of ModifySpeakerCommandsRequest from the provided speakerData.
+     */
+    public Collection<ModifySpeakerCommandsRequest> buildFlowModifyCommands(
+            List<SpeakerData> speakerData, CommandContext context) {
+        return groupBySwitchId(speakerData).entrySet().stream()
+                .map(entry -> {
+                    List<OfCommand> ofCommands = OfCommandConverter.INSTANCE.toOfCommands(entry.getValue());
+                    return buildFlowModifyCommand(entry.getKey(), ofCommands, context);
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Build a ModifySpeakerCommandsRequest from the provided OF commands.
+     */
+    public ModifySpeakerCommandsRequest buildFlowModifyCommand(SwitchId switchId, List<OfCommand> ofCommands,
```

Comment:
I think this function may be private?

Reply:
yes, but I made it public to use same pattern. `buildFlowDeleteCommand` and `buildFlowInstallCommand` also public and we use them outside of this class. `buildFlowModifyCommand` also will be used from different places. I just decided to do not make in private and them make it public back.